cmd/import-validator/core/service.go

```diff
@@ -255,14 +257,31 @@ func checkModifiedState(db *postgres.IndexerDb, l *ledger.Ledger, block *bookkee
 			resourcesForAddress := make(map[ledger.Creatable]ledgercore.AccountResource)
 			resourcesAlgod[address] = resourcesForAddress
 			for creatable := range creatables {
-				var resource ledgercore.AccountResource
-				resource, err1 =
-					l.LookupResource(block.Round(), address, creatable.Index, creatable.Type)
-				if err1 != nil {
-					err1 = fmt.Errorf("checkModifiedState() lookup resource err1: %w", err1)
+				switch creatable.Type {
+				case basics.AssetCreatable:
+					var assetResource ledgercore.AssetResource
+					assetResource, err1 = l.LookupAsset(block.Round(), address, basics.AssetIndex(creatable.Index))
+					if err1 != nil {
+						err1 = fmt.Errorf("checkModifiedState() lookup asset resource err1: %w", err1)
+						return
+					}
+					ar := ledgercore.AccountResource{
+						AssetParams:  assetResource.AssetParams,
+						AssetHolding: assetResource.AssetHolding,
+					}
+					resourcesForAddress[creatable] = ar
+				case basics.AppCreatable:
+					var appResource ledgercore.AppResource
+					appResource, err1 = l.LookupApplication(block.Round(), address, basics.AppIndex(creatable.Index))
+					if err1 != nil {
+						err1 = fmt.Errorf("checkModifiedState() lookup application resource err1: %w", err1)
+						return
+					}
+					resourcesForAddress[creatable] = normalizeAccountResource(appResource)
+				default:
+					err1 = fmt.Errorf("checkModifiedState() unexpected creatable type %v", creatable.Type)
```

Comment:
@tolikzinovyev could you review this section? It seems reasonable, but should we call `normalizeAccountResource(ar)` for the `AssetCreatable` case?

Reply:
I would only suggest to do one of the following:

1. Keep `normalizeAccountResource()` unchanged and call it after the `switch`.
2. Change `normalizeAccountResource()` into `normalizeAppResource()` that would accept and return `ledgercore.AppResource`.

Reply:
Went with changing `normalizeAccountResource` to `convertAppResource` which will return an `AccountResource` with normalized fields. I think it's more properly named and avoids calling the function when it's not necessary for `AssetResource`s.

Reply:
@tolikzinovyev Let me know if you have any feedback on these changes. 😄

Reply:
Looks fine!